GitUI/Editor/CustomSyntaxStrategyManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ICSharpCode.TextEditor.Document;
+
+namespace GitUI.Editor
+{
+    // simple manager which return highighting strategy implemented in current assembly
+    class CustomSyntaxStrategyManager
+    {
+        const string StrategySuffix = "HighlightingStrategy";
+
+        static readonly Dictionary<string, Type> StrategyTypes;
+
+        static CustomSyntaxStrategyManager()
+        {
+            try
+            {
+                StrategyTypes = typeof(CustomSyntaxStrategyManager).Assembly
+                    .GetTypes()
+                    .Where(t => !t.IsAbstract && typeof(IHighlightingStrategy).IsAssignableFrom(t))
+                    .ToDictionary(t => t.Name.EndsWith(StrategySuffix) ? t.Name.Substring(0, t.Name.Length - StrategySuffix.Length).ToUpperInvariant() : t.Name.ToUpperInvariant(), t => t)
+                ;
+            }
+            catch
+            {
+                StrategyTypes = new Dictionary<string, Type>();
+            }
+        }
+
+        public static IHighlightingStrategy TryFindCustomSyntaxStrategy(string syntax)
```

Comment:
personally I think this method name is misleading. 
generally `TryXxx` methods are additions to `Xxx` methods and their intent not to throw during the operations which will result in failures in `Xxx` methods.
they also have signatures similar to the following: `TryXxx(T someValue, out TRequired expectedValue): bool`

this methods doesn't fit in the above description.
perhaps a more correct name could be `GetSyntaxStrategy()` or `CreateSyntaxStrategy()` or even simpler `Create()`